### PR TITLE
FE-2175 GroupedCharacter - added uncontrolled support

### PR DESCRIPTION
**Pull request by mkrds**
# Description
Added support for uncontrolled use of GroupedCharacter component.

### CODE REVIEW

File: src/__experimental__/components/grouped-character/grouped-character.component.js [hunk 1]
```diff
@@ -27,7 +39,10 @@ const GroupedCharacter = (props) => {
 
     const reProcessedInputValue = ev.target.value.split(separator).join('');
 
-    props.onChange({ target: { value: reProcessedInputValue } });
+    onChange({ target: { value: reProcessedInputValue } });
```
Comment (jamime) [inline src/__experimental__/components/grouped-character/grouped-character.component.js]: We need to include the whole `event` in the callback e.g
```js
const target = {
  ...ev.target,
value: reProcessedInputValue
}

ev.target = target;
onChange(ev);
```

Comment (mkrds) [reply src/__experimental__/components/grouped-character/grouped-character.component.js]: How about:
```js
const event= {
  ...ev,
  target: {
    ...ev.target,
    value: reProcessedInputValue
  }
}
onChange(event);
```
No mutations and imo looks more readable.

File: src/__experimental__/components/grouped-character/grouped-character.component.js [hunk 2]
```diff
@@ -27,7 +39,10 @@ const GroupedCharacter = (props) => {
 
     const reProcessedInputValue = ev.target.value.split(separator).join('');
 
-    props.onChange({ target: { value: reProcessedInputValue } });
+    onChange(ev, reProcessedInputValue);
+    if (!isControlled) {
+      setInternalValue(reProcessedInputValue);
+    }
     setTimeout(() => eventRef.setSelectionRange(newCursorPos, newCursorPos));
```
Comment (jamime) [inline src/__experimental__/components/grouped-character/grouped-character.component.js]: `eventRef` is not defined.

File: src/__experimental__/components/grouped-character/grouped-character.component.js [hunk 3]
```diff
@@ -4,12 +4,24 @@ import Textbox from '../textbox';
...
-const GroupedCharacter = (props) => {
+const GroupedCharacter = ({
+  groups,
+  value: externalValue,
+  defaultValue,
+  onChange,
+  separator: rawSeparator,
+  ...rest
+}) => {
   const [pressedKey, updatePressedKey] = useState(null);
```
Comment (jamime) [inline src/__experimental__/components/grouped-character/grouped-character.component.js]: This state is not used

File: change_log/next/FE-2175-grouped-character-uncontrolled-support.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Bug Fixes: "Fixed issue where component did not supporter uncontrolled use. (Components: GroupedCharacter)"
```
Comment (damianOfKyan) [inline change_log/next/FE-2175-grouped-character-uncontrolled-support.yml]: Bug Fixes: "Fixed issue where component did not support uncontrolled use. (Components: GroupedCharacter)"

**Pull request merged**